docs/content/3.composables/1.useCldImageUrl.md

```diff
@@ -0,0 +1,173 @@
+# `useCldImageUrl`
+
+---
+
+This composable is using [@cloudinary-util/url-loader](https://github.com/colbyfayock/cloudinary-util/tree/main/packages/url-loader) under the hood to generate the Cloudinary URL with certain optimizations.
+
+## Usage
+
+The composable allows you to pass optimization options and the src of the image and in return get a full cloudinary url with the optimized images. Then, you can use this url any way you like (for example in the custom image component).
+
+```vue
+<script lang="ts" setup>
+const { url } = useCldImageUrl({
+  options: {
```

Comment:
one thing i did with my similar getCldImageUrl is removed the need to pass options in as an object

https://github.com/colbyfayock/next-cloudinary/blob/main/next-cloudinary/src/helpers/getCldImageUrl.ts#L20

if someone is using this as part of the integration, the config and analytic should be set up already, meaning only thing they should care about is the options as a top level object

the thought is it makes it easier to consume, ill get to the config/analytics later

Reply:
I understand the reasoning behind it but in my imagination I already visualised a community user that would be using this module and he/she would create an issue in this repository that they want to have access to the analytics and why the is this abstraction there ;)

I can make it the way you designed it. You are the author of this package so it is up to you. I am just porting it to Nuxt world ;)

Reply:
to be clear - someone can still access it the way i set it up, i just intentionally made it as 3 top level arguments rather than the first argument taking everything in. it feels easier to use for 99% percent of usecases

so for instance:
```
useCldImageUrl({
  src: 'myid'
})
```

to just use it (most cases)

but if for someone reason someoen wants to pass in their own values, which really there shouldnt be a reason because its purely an internal Cloudinary thing, they could do:

```
useCldImageUrl({
  src: 'myid'
}, config, analytics)
```

the config would be the most reasonable thing to expect someone MIGHT want to configure - but again, most use cases, someone will only be using 1 cloud account which should be globally  configured, so they shouldnt need to ever use that